be/src/exec/json_scanner.cpp

```diff
@@ -0,0 +1,485 @@
+                wbytes = sprintf((char*)tmp_buf, "%lu", value->GetUint64());
+                fill_slot(tuple, desc, tuple_pool, tmp_buf, wbytes);
+            } else if (value->IsInt64()) {
+                wbytes = sprintf((char*)tmp_buf, "%ld", value->GetInt64());
+                fill_slot(tuple, desc, tuple_pool, tmp_buf, wbytes);
+            } else {
+                wbytes = sprintf((char*)tmp_buf, "%f", value->GetDouble());
+                fill_slot(tuple, desc, tuple_pool, tmp_buf, wbytes);
+            }
+            break;
+        case rapidjson::Type::kFalseType:
+            //fill_slot(tuple, desc, tuple_pool, (uint8_t*)"false", 5);
+            fill_slot(tuple, desc, tuple_pool, (uint8_t*)"0", 1);
+            break;
+        case rapidjson::Type::kTrueType:
+            //fill_slot(tuple, desc, tuple_pool, (uint8_t*)"true", 4);
+            fill_slot(tuple, desc, tuple_pool, (uint8_t*)"1", 1);
+            break;
+        default:
+            std::stringstream str_error;
+            str_error << "Invalid JsonType " << value->GetType() << ", Column Name `" << desc->col_name() << "`.";
+            LOG(WARNING) << str_error.str();
+            return Status::RuntimeError(str_error.str());
+    }
+    return Status::OK();
+}
+
+/**
+ * handle input a simple json
+ * For example:
+ *  1. {"doris_data": [{"colunm1":"value1", "colunm2":10}, {"colunm1":"value2", "colunm2":30}]}
+ *  2. {"colunm1":"value1", "colunm2":10}
+ */
+Status JsonReader::handleSimpleJson(Tuple* tuple, std::vector<SlotDescriptor*> slot_descs, MemPool* tuple_pool, bool* eof) {
+    if (_next_line >= _total_lines) {
+        // generic document
+        RETURN_IF_ERROR(parseJsonDoc(eof));
+        if (*eof) {// handle over all data
+            return Status::OK();
+        }
+        if (_jsonDoc.HasMember("doris_data") && _jsonDoc["doris_data"].IsArray() ) {
+            _total_lines = _jsonDoc["doris_data"].Size();
+        } else {
+            _total_lines = 1;
+        }
+        _next_line = 0;
+    }
+
+    if (_jsonDoc.HasMember("doris_data") && _jsonDoc["doris_data"].IsArray()) {//handle case 1
+        rapidjson::Value& valueArray = _jsonDoc["doris_data"];
+        rapidjson::Value& objectValue = valueArray[_next_line++];// json object
+        for (auto v : slot_descs) {
+            if (objectValue.HasMember(v->col_name().c_str())) {
+                rapidjson::Value& value = objectValue[v->col_name().c_str()];
+                writeDataToTuple(&value, v, tuple, tuple_pool);
+            } else {
+                if (v->is_nullable()) {
+                    tuple->set_null(v->null_indicator_offset());
+                } else  {
+                    std::stringstream str_error;
+                    str_error << "The column `" << v->col_name() << "` is not nullable, but it's not found in jsondata.";
+                    LOG(WARNING) << str_error.str();
+                    return Status::RuntimeError(str_error.str());
```

Comment:
Here you directly return an error, and the entire load job will be failed.
But sometimes user will set the `max_filter_ratio` to try to skip some error line.